src/main/java/de/maxhenkel/audioplayer/FilenameMappings.java

```diff
@@ -0,0 +1,87 @@
+package de.maxhenkel.audioplayer;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.UUID;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.NbtAccounter;
+import net.minecraft.nbt.NbtIo;
+import net.minecraft.server.MinecraftServer;
+
+public class FilenameMappings {
+
+    public static void append(MinecraftServer server, String filename, UUID sound) {
+        URI location = server.getWorldPath(AudioManager.AUDIO_DATA).resolve("filename_mappings.dat").toUri();
+        File mappingsFile = new File(location);
+        String extension = filename.substring(filename.length() - 4);
+
+        try {
+            CompoundTag root = loadOrDefault(mappingsFile);
+            ListTag mappings = getMappingListOrDefault(root);
+
+            CompoundTag entry = new CompoundTag();
+            entry.putString("uuid", sound.toString() + extension);
+            entry.putString("filename", filename);
+
+            mappings.add(entry);
+
+            if (!root.contains("mappings")) {
+                root.put("mappings", mappings);
+            }
+
+            NbtIo.writeCompressed(root, mappingsFile.toPath());
+        } catch (IOException e) {
+            System.err.println("Permission Denied. Failed to read or write the file.");
+        }
+
+        verifyFilesExist(server);
+    }
+
+    public static void verifyFilesExist(MinecraftServer server) {
+        URI location = server.getWorldPath(AudioManager.AUDIO_DATA).resolve("filename_mappings.dat").toUri();
+        File mappingsFile = new File(location);
+
+        try {
+            CompoundTag root = loadOrDefault(mappingsFile);
+            ListTag mappings = getMappingListOrDefault(root);
+
+            for (int i = 0; i < mappings.size(); i++) {
+                String uuid = mappings.getCompound(i).getString("uuid");
+                URI filepath = server.getWorldPath(AudioManager.AUDIO_DATA).resolve(uuid).toUri();
+
+                if (!new File(filepath).exists()) {
+                    mappings.remove(i);
+                }
+            }
+
+            NbtIo.writeCompressed(root, mappingsFile.toPath());
+
+            if (mappings.isEmpty()) {
+                mappingsFile.delete();
+            }
+        } catch (IOException e) {
+            System.err.println("Permission Denied. Failed to read or write the file.");
+        }
+    }
+
+    public static void autoSaveHappened() {
+        AudioPlayer.LOGGER.warn("Auto Save Triggered!");
```

Comment:
Why is this needed??

Reply:
Oh, sorry thats not. I must of forgot to remove that, I was tested to see if i could auto check the mappings when a auto save happened but i couldn't find a way to detected that event.